packages/jest-util/src/deep_cyclic_copy.js

```diff
@@ -71,16 +80,20 @@ function deepCyclicCopyObject(
 
 function deepCyclicCopyArray(
   array: Array<any>,
-  blacklist: Set<string>,
+  options: DeepCyclicCopyOptions,
   cycles: WeakMap<any, any>,
 ): Array<any> {
-  const newArray = [];
+  const newArray = options.keepPrototype
+    ? // $FlowFixMe: getPrototypeOf an array is OK.
+      new (Object.getPrototypeOf(array)).constructor(array.length)
+    : [];
   const length = array.length;
 
   cycles.set(array, newArray);
 
   for (let i = 0; i < length; i++) {
-    newArray[i] = deepCyclicCopy(array[i], EMPTY, cycles);
+    delete options.blacklist;
```

Comment:
We should be good moving this out of the loop

Reply:
Woah, this is a bug -- it deletes the blacklist on the first key iteration so only the first key can be blacklisted